lib/Conversion/TorchToTosa/TosaLegalizeCommon.cpp

```diff
@@ -268,7 +273,8 @@ convertReduceMeanOp(PatternRewriter &rewriter, Operation *op,
     int64_t axis_val = axes_elems.getValues<IntegerAttr>()[i].getInt();
     if (axis_val < 0)
       axis_val += input_rank;
-    num_elems_on_reduced_axis *= input_type.getShape()[axis_val];
+    num_elems_on_reduced_axis *=
+        torch::Torch::makeShapeTorchCompatible(input_type.getShape())[axis_val];
```

Comment:
@eric-k256 -- TosaLegalizeCommon is [supposed to be upstream](https://discourse.llvm.org/t/rfc-moving-tosa-construction-lib-into-mlir-core/4961) -- so it doesn't make sense for it to call Torch-MLIR-specific helpers. Just noting that this is causing divergence and tech debt -- it would be nice if you folks could actually move this upstream as was intended.

Reply:
Yes, a copy in here would be good.

Reply:
Yes, this is meant to move upstream, but got a bit lost with Suraj leaving. I'll try to bring it back, but will take some work on the test frontend before we can accomplish it.
In the mean time, for this case, rather than multiplying here, if a dimension is unknown / dynamic, I would return immediately. Otherwise you are creating ReduceSum(t) and multiplying the result by a garbage value. The failure lets us know that we are failing due to an unknown number of elements to divide by, rather than silently outputting garbage.